zero/dg_iz.c

```diff
@@ -214,24 +211,29 @@ void gkyl_dg_iz_coll(const struct gkyl_dg_iz *up, const struct gkyl_array *moms_
     cell_center = (m0_idx - 0.5)*up->dlogM0 + up->minLogM0;
     cell_vals_2d[1] = 2.0*(log_m0_av - cell_center)/up->dlogM0; // M0 value on cell interval
 
-    if ((up->E/temp_elc_av >= 3./2.) || (m0_elc_av <= 0.) || (temp_elc_av <= 0.)) {
+    if ((m0_elc_av <= 0.) || (temp_elc_av <= 0.)) {
       coef_iz_d[0] = 0.0;
     }
     else {
       double *iz_dat_d = gkyl_array_fetch(up->ioniz_data, gkyl_range_idx(&up->adas_rng, (int[2]) {t_idx,m0_idx}));
       double adas_eval = up->adas_basis.eval_expand(cell_vals_2d, iz_dat_d);
       coef_iz_d[0] = pow(10.0,adas_eval)/cell_av_fac;
+
+      // calculate vtSq_iz at each cell
+      if (up->E/temp_elc_av >= 3./2.) {
+      	// calculate vtSq_iz using current model
+      	// vtSq_iz = vtSq_elc/2.0 - Eiz/(3*me)
+      	array_set1(nc, vtSq_iz_d, 0.5, vtSq_elc_d);
```

Comment:
@tnbernard i would just use `vtSq_iz->ncomp` instead of `nc` here, since this is it's only use and it doesn't cost anything (and you can write the GPU code similarly).